force-app/main/default/classes/LeadTriggerTest.cls

```diff
@@ -0,0 +1,312 @@
+@IsTest
+public with sharing class LeadTriggerTest {
+
+    @TestSetup
+    static void setUpData() {
+
+        // create test account
+        Account testAccount = new Account(Name='Test Account');
+        insert testAccount;
+    
+        // create test contact
+        Contact testContact = new Contact(FirstName='Test', LastName='Contact', Email='test@example.com', AccountId=testAccount.Id);
+        insert testContact;
+    }
+
+    /*
+    * Test Title of a Lead has been update to Vice President, Manager, Assistant, Executive.
+    */
+    @IsTest
+    static void testHandleTitleNormalization_Insert() {
+
+        // creates leads        
+        List<Lead> ldList = new List<Lead>();
+
+        Lead leadVP = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'v.p.');
+        Lead leadMgr = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'manage');
+        Lead leadExec = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'chief');
+        Lead leadAssist = new Lead(FirstName = 'Test', LastName = 'Lead', Company = 'Test Company', Title = 'deputy');
+
+        ldList.add(leadVP);
+        ldList.add(leadMgr);
+        ldList.add(leadExec);
+        ldList.add(leadAssist);
+
+        // perform test
+        Test.startTest();
+        insert ldList;
+        Test.stopTest();
+
+        // get values to check things are working as they should
+        List<Lead> ldListCheckVP = [select Id, Title from Lead where Id = :leadVP.Id];
+        List<Lead> ldListCheckMgr = [select Id, Title from Lead where Id = :leadMgr.Id];
+        List<Lead> ldListCheckExec = [select Id, Title from Lead where Id = :leadExec.Id];
+        List<Lead> ldListCheckAssist = [select Id, Title from Lead where Id = :leadAssist.Id];
+
+        // check results
+        System.assertEquals('Vice President', ldListCheckVP[0].Title, 'VP Title is not Vice President');
+        System.assertEquals('Manager', ldListCheckMgr[0].Title, 'Mgr Title is not Manager');
+        System.assertEquals('Executive', ldListCheckExec[0].Title, 'Exec Title is not Executive');
+        System.assertEquals('Assistant', ldListCheckAssist[0].Title, 'Assistant Title is not Assistant');
+    }
+
+    /*
+    * Test updating the titles to make sure the before update trigger fires and test pass.
+    */
+    @IsTest
+    static void testHandleTitleNormalization_Update() {
+
+        // creates leads        
+        List<Lead> ldList = new List<Lead>();
+
+        Lead leadVP = new Lead(FirstName='Test', LastName='Lead', Company='Test Company', Title='v.p.');
```

Comment:
Since you are inserting multiple leads in each test it might save you some repetition if you create a helper method to build the records.